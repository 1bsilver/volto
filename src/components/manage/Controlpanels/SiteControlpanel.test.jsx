import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-intl-redux';
import { MemoryRouter, Route } from 'react-router-dom';

import SiteControlpanel from './SiteControlpanel';

const mockStore = configureStore();

jest.mock('react-portal', () => ({
  Portal: jest.fn(() => <div id="Portal" />),
}));
jest.mock('../Form/Form', () => jest.fn(() => <div id="form" />));

describe('SiteControlpanel', () => {
  it('renders a site setup controlpanel component', () => {
    const store = mockStore({
      controlpanels: {
        controlpanel: {
          '@id': 'http://localhost:8080/Plone/@controlpanels/site',
          title: 'Website',
          schema: {
            fieldsets: [],
            properties: [],
          },
          data: {},
        },
        update: {
          loading: false,
          loaded: true,
        },
      },
      intl: {
        locale: 'en',
        messages: {},
      },
    });
    const component = renderer.create(
      <Provider store={store}>
        <MemoryRouter initialEntries={['/controlpanel/site']}>
          <Route path={'/controlpanel/:id'} component={SiteControlpanel} />
        </MemoryRouter>
      </Provider>,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
