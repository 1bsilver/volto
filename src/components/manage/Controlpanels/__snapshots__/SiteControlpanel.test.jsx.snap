// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SiteControlpanel renders a site setup controlpanel component 1`] = `
<div
  id="page-controlpanel"
>
  <div
    className="ui container"
  >
    <div
      id="form"
    />
  </div>
  <div
    id="Portal"
  />
</div>
`;
